src/common/common_test.go

```diff
@@ -182,51 +183,148 @@ var _ = Describe("verify common", func() {
 			Expect(removeUninitializedTaint).To(BeFalse())
 		})
 
 		for _, test := range []struct {
 			PlatformType                     models.PlatformType
 			ExpectedRemoveUninitializedTaint bool
+			Invoker                          string
+			HasValidvSphereCredentials       bool
+			VersionOpenshift                 string
 		}{
 			{
 				PlatformType:                     models.PlatformTypeNutanix,
+				Invoker:                          InvokerAssisted,
+				VersionOpenshift:                 "4.14.0",
 				ExpectedRemoveUninitializedTaint: true,
 			},
 			{
 				PlatformType:                     models.PlatformTypeVsphere,
+				Invoker:                          InvokerAssisted,
+				HasValidvSphereCredentials:       false,
+				VersionOpenshift:                 "4.14.0-rc0",
 				ExpectedRemoveUninitializedTaint: true,
 			},
+			{
+				PlatformType:                     models.PlatformTypeVsphere,
+				Invoker:                          InvokerAgent,
+				HasValidvSphereCredentials:       false,
+				VersionOpenshift:                 "4.14.0-rc0",
+				ExpectedRemoveUninitializedTaint: true,
+			},
+			{
+				PlatformType:                     models.PlatformTypeVsphere,
+				Invoker:                          InvokerAgent,
+				HasValidvSphereCredentials:       false,
+				VersionOpenshift:                 "4.15.0",
+				ExpectedRemoveUninitializedTaint: true,
+			},
+			{
+				PlatformType:                     models.PlatformTypeVsphere,
+				Invoker:                          InvokerAgent,
+				HasValidvSphereCredentials:       true,
+				VersionOpenshift:                 "4.14.0",
+				ExpectedRemoveUninitializedTaint: true,
+			},
+			{
+				PlatformType:                     models.PlatformTypeVsphere,
+				Invoker:                          InvokerAgent,
+				HasValidvSphereCredentials:       true,
+				VersionOpenshift:                 "4.15",
+				ExpectedRemoveUninitializedTaint: false,
+			},
 			{
 				PlatformType:                     models.PlatformTypeNone,
+				Invoker:                          InvokerAssisted,
+				VersionOpenshift:                 "4.14.0-rc10-z10",
 				ExpectedRemoveUninitializedTaint: false,
 			},
 			{
 				PlatformType:                     models.PlatformTypeBaremetal,
+				Invoker:                          InvokerAssisted,
+				VersionOpenshift:                 "4.14.0",
 				ExpectedRemoveUninitializedTaint: false,
 			},
 		} {
-			platformType := test.PlatformType
-			expectedRemoveUninitializedTaint := test.ExpectedRemoveUninitializedTaint
-
 			Context("by platform", func() {
-				var platform *models.Platform
-				BeforeEach(func() {
-					platform = &models.Platform{
-						Type: &platformType,
+				It(fmt.Sprintf("platform %v, invoker %v, hasValidCredentials %v, version %v, expected remove uninitialized taint = %v", test.PlatformType, test.Invoker, test.HasValidvSphereCredentials, test.VersionOpenshift, test.ExpectedRemoveUninitializedTaint), func() {
+					platform := &models.Platform{
```

Comment:
```suggestion
					platform := &models.Platform{
```

does this test loop works as expected? usually we need to assign test to some variable before using it because of the way ginkgo works. I suggest to move this test to use `DescribeTable` to avoid such issue.

Reply:
It didn't work as expected. Thanks for pointing out this issue. It has been fixed.